packages/sdk/src/protocols/filter.ts

```diff
@@ -261,26 +261,34 @@ class FilterSDK extends BaseProtocolSDK implements IFilterSDK {
   public readonly protocol: FilterCore;
 
   private activeSubscriptions = new Map<string, SubscriptionManager>();
-  private async handleIncomingMessage(
-    pubsubTopic: PubsubTopic,
-    wakuMessage: WakuMessage
-  ): Promise<void> {
-    const subscription = this.getActiveSubscription(pubsubTopic);
-    if (!subscription) {
-      log.error(`No subscription locally registered for topic ${pubsubTopic}`);
-      return;
-    }
-
-    await subscription.processIncomingMessage(wakuMessage);
-  }
 
-  constructor(libp2p: Libp2p, options?: ProtocolCreateOptions) {
-    super({ numPeersToUse: options?.numPeersToUse });
-    this.protocol = new FilterCore(
-      this.handleIncomingMessage.bind(this),
-      libp2p,
-      options
+  constructor(
+    connectionManager: ConnectionManager,
+    libp2p: Libp2p,
+    options?: ProtocolCreateOptions
+  ) {
+    super(
```

Comment:
it seems there is a room for improvement in the class structure as it becomes harder to understand
do you think we can do `SDK -> Core -> BaseProtocol` easily or it would require major refactoring?

Reply:
good, captured it here https://github.com/waku-org/js-waku/issues/2033